### ISSUE TITLE
consult-ine does not unfold org subtree

### ISSUE THREAD

**Issue by armindarvish** [minad/consult#674]
I am using consult-line and have noticed that in org-mode when I search for a word that is under a folded subtree, it does not automatically unfold the subtree after I select that line. 

For example, in the screenshots below, I have "consult" as a subheader buried in a few headers under Emacs (SC1). If everything is unfolded, and I search with consult-line to find "consult". I can see the consult subheader in preview (SC2) but the actual buffer remain unfolded. After I select that line it will jump to the first unfolded header which is "Emacs" which is not the right line to go to!

Ideally, it might be useful to have both functionality, where a default selection automatically goes to the right unfolded line and an alternative selection (like C-Ret or M-Ret) goes to the unfolded header. 

SC1: The tree structure Emacs>UI>Completion>Consult:
<img width="1461" alt="image" src="https://user-images.githubusercontent.com/18317557/196292071-e4bbaae5-4791-4b02-bbf3-c632f932e7f4.png">

SC2: using consult-line when everything is unfolded with preview
<img width="1465" alt="image" src="https://user-images.githubusercontent.com/18317557/196292387-e8f9a69c-441b-4154-9a99-e079fe38c808.png">

SC3: after I select the result and jump
<img width="1467" alt="image" src="https://user-images.githubusercontent.com/18317557/196292429-4a749042-a95a-47d5-b440-fc77c1be2276.png">

**Comment by minad** [comment]
Probably a duplicate of #563? I'd need more informatiod, see https://github.com/minad/consult#bug-reports.

**Comment by armindarvish** [comment]
@minad just FYI, I went through your bug report process and found out that in my config I had a line where I was using (straight-use-package 'org) to update orgmode to the latest version and that was breaking the unfolding. This was very confusing because when I checked the Org version it is still 9.5.5 but apparently straight is pulling some newer commits that is causing the issue. With the built-in Org 9.5.5 in my emacs 29 consult-line works as expected. 

That said, my emacs might be behind. I'm on MacOS and I installed emacs with brew a few months ago and have not updated it for a while.  So, this problem might show up again if those changes merge into the built-in orgmode in the future. Interestingly, other similar packages like swiper still worked just fine even with the latest orgmode, and that was a source of confusion because I did not realize that the orgmode is the source of the issue!
